js/index.js

```diff
@@ -38,5 +40,94 @@ const siteContent = {
 };
 
 // Example: Update the img src for the logo
+
+//nav selector
+const [ nav ] = document.getElementsByTagName('nav');
 let logo = document.getElementById("logo-img");
-logo.setAttribute('src', siteContent["nav"]["img-src"])
+
+// CTA Selector
+const [ cta ] = document.getElementsByClassName('cta');
+const [ ctaH1 ] = document.getElementsByTagName('h1');
+const [ ctaBtn ] = document.getElementsByTagName('button');
+const ctaImg = document.getElementById('cta-img');
+
+//Main Content Selector
+const [ mc ] = document.getElementsByClassName('main-content');
+
+//Middle Image Selector
+const middleImg = document.getElementById('middle-img');
+
+//Bottom Content Selector
+const [ bc ] = document.getElementsByClassName("bottom-content");
+
+// Text Content H4 and P Content Selectors
+const textContentH4 = document.querySelectorAll(".text-content h4");
+const textContentP = document.querySelectorAll(".text-content p");
+
+// Contact H4 and P Content Selectors
+const contactH4 = document.querySelectorAll(".contact h4");
+const contactP = document.querySelectorAll(".contact p");
+
+// Footer
+const footerP = document.querySelectorAll("footer p");
+
+// Mutate the Dom
+
+//Header
+logo.setAttribute('src', siteContent["nav"]["img-src"]);
+
+var newLink1 = document.createElement("a");
+newLink1.textContent = siteContent["nav"]["nav-item-7"];
+
+var newLink2 = document.createElement("a");
+newLink2.textContent = siteContent["nav"]["nav-item-8"];
+
+nav.appendChild(newLink1);
+nav.prepend(newLink2);
+
+Array.from(nav.children).forEach((value, index) => {
+  value.textContent = siteContent["nav"][`nav-item-${index + 1}`];
+  value.setAttribute('style', "color:green;");
+});
+
+
+
+//CTA
+ctaH1.textContent = siteContent["cta"]["h1"];
+ctaImg.setAttribute('src', siteContent["cta"]["img-src"]);
+ctaBtn.textContent = siteContent["cta"]["button"];
+
+//Main Content
+
+//Top Content
+textContentH4[0].textContent = siteContent["main-content"]["features-h4"];
+textContentP[0].textContent = siteContent["main-content"]["features-content"];
+
+textContentH4[1].textContent = siteContent["main-content"]["about-h4"];
+textContentP[1].textContent = siteContent["main-content"]["about-content"];
+
+
+//Middle Image
+middleImg.setAttribute('src', siteContent["main-content"]["middle-img-src"]);
+
+//Bottom Content
+textContentH4[2].textContent = siteContent["main-content"]["services-h4"];
+textContentP[2].textContent = siteContent["main-content"]["services-content"];
```

Comment:
So, the way you're assigning the headers and content here are very nice and to the point, but imagine if we had, say... a hundred titles and paragraphs! That'd be a lot of assignments and repeated code.

So how could we make it more DRY?

Well, I see an opportunity here for loops! We just need to make sure that we're referring to the correct siteContent values for each `h4` and `content` key. If we check out siteContent, we can see that the order in which our keys show up are the same as how they are presented on the DOM.

So, we could just grab the h4 items in an array and map the indexes to h4 nodes we gathered.

For example:
```js
// first, we grab all the `siteContent` keys into an array
const mainContentKeys = Object.keys(siteContent['main-content']);
// now we need to grab the h4 keys specifically - luckily every h4 element ends with the string 'h4'
const mainContentH4s = mainContentKeys.filter(key => key.endsWith('h4'));
// and finally, we can loop through our h4 elements and set them using the equivalent index of our `textContentH$` array
mainContentH4s.forEach((value, idx) => {
  textContentH4[idx].textContent = siteContent['main-content'][value];
});
```